src/core_plugins/kibana/public/management/sections/indices/_edit.js

```diff
@@ -70,23 +71,37 @@ uiModules.get('apps/management')
   });
 
   $scope.refreshFields = function () {
-    $scope.indexPattern.refreshFields();
+    const confirmModalOptions = {
+      confirmButtonText: 'Refresh fields',
+      onConfirm: () => { $scope.indexPattern.refreshFields(); }
+    };
+    confirmModal(
+      'This will reset the field popularity counters. Are you sure you want to refresh your fields?',
+      confirmModalOptions);
```

Comment:
AirBnB style guide suggests putting this closing parens on its own line: https://github.com/airbnb/javascript#functions--signature-invocation-indentation